astrosat_users/views/views_profiles.py

```diff
@@ -0,0 +1,53 @@
+from django.contrib.auth import get_user_model
+from django.shortcuts import get_object_or_404
+from django.utils.functional import cached_property
+
+from rest_framework import generics
+from rest_framework.exceptions import APIException
+from rest_framework.permissions import IsAuthenticated, BasePermission
+
+from astrosat.decorators import swagger_fake
+
+
+class NoMatchingProfileException(APIException):
+    status_code = 400
+    default_detail = "Unable to find a profile for the given user"
+    default_code = "no_matching_profile"
+
+
+class IsAdminOrSelf(BasePermission):
+    def has_permission(self, request, view):
+        user = request.user
+        return user.is_superuser or user == view.user
+
+
+class UserProfileView(generics.RetrieveUpdateAPIView):
```

Comment:
It's done via signals: https://github.com/astrosat/orbis/blob/881fbbaac53533633d2d3ba36df6d9e687e5fc36/server/orbis/signals/signals_users.py#L8

Reply:
Since we are adding this when we already have users, I assume this still works. The comment in that file states. 

>If a User has just been created, then the corresponding ORBIS Profile must also be created

It doesn't say anything about users that already exist.